src/libostree/ostree-repo-pull.c

```diff
@@ -871,7 +871,12 @@ content_fetch_on_complete (GObject        *object,
   checksum_obj = ostree_object_to_string (checksum, objtype);
   g_debug ("fetch of %s complete", checksum_obj);
 
-  if (pull_data->is_mirror && pull_data->repo->mode == OSTREE_REPO_MODE_ARCHIVE_Z2)
+  /* If we're mirroring and writing into an archive repo, we can directly copy
+   * the content rather than paying the cost of exploding it, checksumming, and
+   * re-gzip.
+   */
```

Comment:
`pull --mirror` into `archive` repos, yes.  That's currently intentional, since if we want to do mirroring of static deltas we'll basically need that.  Though if someone wanted to change that we could try to validate, but it'd get very expensive for deltas at least.